src/tree/param.cc

```diff
@@ -0,0 +1,79 @@
+/*!
+ * Copyright by Contributors 2019
+ */
+#include <iostream>
+#include <vector>
+
+#include "param.h"
+
+namespace std {
+std::istream &operator>>(std::istream &is, std::vector<int> &t) {
+  // get (
+  while (true) {
+    char ch = is.peek();
+    if (isdigit(ch)) {
+      int idx;
+      if (is >> idx) {
+        t.assign(&idx, &idx + 1);
```

Comment:
Just curious, is this better than `t = { idx };`?

Reply:
Honestly I don't know what assign does.  It's simply old code moved to a cc file to avoid recompiling half of XGBoost when debugging.  I only added a few lines handling empty vector.

Reply:
Got it. In that case, let’s use `=` operator for the sake of clarity.

Reply:
@hcho3 Used `emplace_back` with `clear` added at the beginning.